ext/lib/crypto/mbedtls/CMakeLists.txt

```diff
@@ -10,6 +10,9 @@ if(CONFIG_MBEDTLS_BUILTIN)
 	configs
 	)
 
+  zephyr_include_directories(include)
+  zephyr_include_directories(configs)
+
```

Comment:
This prevents first-party projects from minimizing their include directory scope and therefore avoiding aliasing issues.

I assume the third-party project is using "zephyr_get_include_directories_for_lang" to extract these include paths?
I would recommend either hardcoding ${ZEPHYR_BASE}/ext/lib/crypto/mbedtls/include, or extracting these includes from the "mbedTLS" target just as "zephyr_get_include_directories_for_lang" extracts the includes from the "zephyr_interface" target.

Reply:
> This prevents first-party projects from minimizing their include directory scope and therefore avoiding aliasing issues.

1. But it should happen only if CONFIG_MBEDTLS is enabled.
2. There're definitely more third-party, than first-party, projects, and they're not less important.

> I assume the third-party project is using "zephyr_get_include_directories_for_lang" to extract these include paths?

Yes. More specifically, 3rd-party project calls "make outputexports", and that's how outputexports is implemented. 3rd-party project are however not concerned with internal implementation of outputexports, they just need it to behave thoroughly and faithfully. (Unfortunately, outputexports support regressed after migration of Zephyr build system to cmake, and isn't fully recovered, and parts of its support need to be worked around in each and every 3rd-party project, which approach starts to show its non-scalability).

> I would recommend either hardcoding ${ZEPHYR_BASE}/ext/lib/crypto/mbedtls/include

Hardcoding where?

> or extracting these includes from the "mbedTLS" target just as "zephyr_get_include_directories_for_lang" extracts the includes from the "zephyr_interface" target.

Then we need to recover "outputexports" target to how it was before cmake, and then these cmake tricks need to be implemented - once, not hit again and again by various 3rd-party projects.

Reply:
> Unfortunately, outputexports support regressed after migration of Zephyr build system to cmake

That's https://github.com/zephyrproject-rtos/zephyr/issues/4917

Reply:
> But it should happen only if CONFIG_MBEDTLS is enabled.

Not sure what you mean, but when CONFIG_MBEDTLS is enabled, we still don't want all source files to be built with the mbedTLS include path, just the sources that use mbedTLS.

> Hardcoding where?

The system that is extracting and adding Zephyr include paths I assume is also able to statically add an include path. For instance, like how we statically add OpenThread include paths into Zephyr from the external build system of OpenThread:

https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/openthread/CMakeLists.txt#L208

A quality contribution that resolves #4917 would be welcome.